### PR TITLE
resolve: Do not skip extern prelude during speculative resolution

### PR DESCRIPTION
**Pull request by petrochenkov**
Fixes https://github.com/rust-lang/rust/issues/54665

### CODE REVIEW

File: src/test/run-pass/extern/extern-prelude-no-speculative.rs [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 // run-pass
 #![allow(unused_variables)]
-// compile-flags: --extern LooksLikeExternCrate=/path/to/nowhere
+// compile-flags: --extern LooksLikeExternCrate
```
Comment (petrochenkov) [inline src/test/run-pass/extern/extern-prelude-no-speculative.rs]: cc https://github.com/rust-lang/rust/issues/55103

**Comment by petrochenkov** [comment]
I've added beta nomination just in case, since this affects anything using extern prelude and extern prelude is stabilized on beta.

**Review by nikomatsakis — APPROVED** [review]

**Comment by petrochenkov** [comment]
@bors p=1
(Because beta-nominated and the release is soon.)

**Comment by pnkfelix** [comment]
discussed at compiler team meeting. beta-accepted.

**Pull request merged**